### PR TITLE
templates: Migrate from Container Linux to Fedora CoreOS config format

### PR DESCRIPTION
**Pull request by LorbusChris**
Migrate from Container Linux to Fedora CoreOS config format
~~Right now this retains support for CL configs although I think this might not be necessary.~~

@openshift/openshift-team-mco

~~Blocked on: https://github.com/openshift/machine-config-operator/pull/1677~~
/hold
until master/4.6 opens

**- What I did**
- templates: Merge MCD host service and dropin into one config
- templates: Name all systemd unit configs *.service.yaml
- pkgs/controller: Add support for transpiling Fedora CoreOS configs
Adds support to the template renderer for transpiling Fedora CoreOS
configs to Ignition spec v2 config
- Update vendor
- templates: Migrate to FC config format
Drops the the `filesystem` field on all templates to make them
adhere to FC config format.

**- How to verify it**
CI e2e

**- Description for the changelog**
templates: Migrate to FC config format

### CODE REVIEW

File: pkg/controller/template/render_test.go [hunk 1]
```diff
@@ -230,8 +230,8 @@ func TestSkipMissing(t *testing.T) {
 }
 
 const (
-	templateDir = "../../../templates"
-	resultDir   = "./test_data/templates"
```
Comment (LorbusChris) [inline pkg/controller/template/render_test.go:234]: resultDir wasn't used anywhere afaict

File: pkg/controller/template/render.go [hunk 2]
```diff
@@ -266,9 +262,9 @@ func generateMachineConfigForName(config *RenderConfig, role, name, templateDir,
 		return vs
 	}
 
-	ignCfg, err := transpileToIgn(keySortVals(files), keySortVals(units))
+	ignCfg, err := ctrlcommon.TranspileToIgn2Wrapper(keySortVals(files), keySortVals(units))
 	if err != nil {
-		return nil, fmt.Errorf("error transpiling ct config to Ignition config: %v", err)
+		return nil, fmt.Errorf("error transpiling CT or FC config to Ignition config: %v", err)
```
Comment (LorbusChris) [inline pkg/controller/template/render.go:267]: nit: this should read `CL or FC` or `Container Linux or Fedora CoreOS`

**Comment by runcom** [comment]
Awesome

/approve

**Comment by LorbusChris** [comment]
I think I should probably remove CL config support altogether here. Nobody is going to ever run MCO with CL templates after this again, and there's no support for a mixed set of configs in this PR anyway (so it would have to be **CL-only** templates)

**Comment by LorbusChris** [comment]
Requires: https://github.com/openshift/machine-config-operator/pull/1755

**Comment by LorbusChris** [comment]
Lifting the hold now that 4.6 is open
/hold cancel

**Comment by cgwalters** [comment]
Took a quick look, I think these are known flakes.

**Pull request merged**